### PR TITLE
Improvements around aliased types

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Issue a diagnostic error if an exported, aliased type contains nested
types that aren't exported from the same package.
Include methods for aliased types in the source package.
Removed references to ioutil as it's been deprecated.

### CODE REVIEW

File: src/go/cmd/content.go [hunk 1]
```diff
@@ -323,6 +323,26 @@ func (c *content) filterDeclarations(typ string, decls map[string]Declaration, t
 	return results
 }
 
+// findMethods takes the name of the receiver and looks for Funcs that are methods on that receiver.
+func (c *content) findMethods(s string) map[string]Func {
```
Comment (chlowell) [inline src/go/cmd/content.go:327]: This looks a slightly modified `searchForMethods`. If we need both, can one call the other?

**Review by chlowell — COMMENTED** [review]
Is it feasible to add a unit test for this?

File: src/go/cmd/content.go (hunk 1 above)
Comment (jhendrixMSFT) [reply src/go/cmd/content.go:327]: That was my original intention.  However, `searchForMethods` does a bit more work when ranging over `c.Funcs` so I left them split.  Still, they could be consolidated with a little extra work.

File: src/go/cmd/module.go [hunk 2]
```diff
@@ -120,15 +122,40 @@ func NewModule(dir string) (*Module, error) {
...
+					hoistMethodsForType(source, alias, p)
 				default:
 					fmt.Printf("unexpected node type %T\n", def.n.Type)
 					t = p.c.addSimpleType(*p, alias, p.Name(), originalName)
 				}
 			} else {
 				fmt.Println("found no definition for " + qn)
 			}
+
+			if aliasedStruct != nil {
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply src/go/cmd/module.go:138]: Good call, fixed.

File: src/go/cmd/content.go (hunk 1 above)
Comment (jhendrixMSFT) [reply src/go/cmd/content.go:327]: Fixed.

**Comment by jhendrixMSFT** [comment]
@chlowell added some unit tests

**Review by chlowell — APPROVED** [review]

**Pull request merged**